thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionTaskRunner.java

```diff
@@ -91,6 +98,10 @@ public DetectionTaskRunner() {
     Map<DimensionKey, MetricTimeSeries> dimensionKeyMetricTimeSeriesMap =
         TimeSeriesUtil.getTimeSeriesForAnomalyDetection(anomalyFunctionSpec, startEndTimeRanges);
 
+    overrideConfigs = OverrideConfigHelper.getTimeSeriesOverrideConfigs(
```

Comment:
It is not making query for each time bucket. It is making queries for finding the override config that overlaps the window of current time range and all baseline time ranges. So, there are only 4 queries if we have 1 current time window and 3 baseline weeks.